src/platform/web/ui/session/room/timeline/TextMessageView.js

```diff
@@ -24,19 +25,37 @@ export class TextMessageView extends BaseMessageView {
             className: {
                 "Timeline_messageBody": true,
                 statusMessage: vm => vm.shape === "message-status",
-            },
-        });
+            }
+        }, t.mapView(vm => vm.replyTextTile, replyTextTile => {
+            if (vm.isReply && !replyTextTile) {
+                return new ReplyPreviewError();
+            }
+            else if (replyTextTile && !this._isReplyPreview) {
+                // if this._isReplyPreview = true, this is already a reply preview, don't nest replies for now.
+                return new ReplyPreviewView(replyTextTile);
+            }
+            else {
+                return null;
+            }
+        }));
+
         t.mapSideEffect(vm => vm.body, body => {
-            while (container.lastChild) {
+            while (this._shouldRemove(container.lastChild)) {
                 container.removeChild(container.lastChild);
             }
             for (const part of body.parts) {
                 container.appendChild(renderPart(part));
             }
             container.appendChild(time);
         });
+
         return container;
     }
+
+    _shouldRemove(element) {
+        return element && element.className !== "ReplyPreviewView" && element.nodeName !== "#comment";
```

Comment:
I'd rather exclude any node type not an element
```suggestion
        return element?.nodeType === Node.ELEMENT_NODE && element.className !== "ReplyPreviewView";
```